packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/gollm/GoLLMController.java

```diff
@@ -716,6 +724,109 @@ public ResponseEntity<TaskResponse> createEnrichAMRTask(
 		return ResponseEntity.ok().body(resp);
 	}
 
+	@PostMapping("/equations-from-image")
+	@Secured(Roles.USER)
+	@Operation(summary = "Dispatch a `GoLLM Equations from image` task")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "200",
+				description = "Dispatched successfully",
+				content = @Content(
+					mediaType = "application/json",
+					schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = TaskResponse.class)
+				)
+			),
+			@ApiResponse(responseCode = "400", description = "The provided image is invalid", content = @Content),
+			@ApiResponse(responseCode = "500", description = "There was an issue dispatching the request", content = @Content)
+		}
+	)
+	public ResponseEntity<TaskResponse> equationsFromImageTask(
+		@RequestParam(name = "project-id", required = false) final UUID projectId,
+		@RequestParam(name = "document-id", required = true) final UUID documentId,
+		@RequestParam(name = "mode", required = false, defaultValue = "ASYNC") final TaskMode mode,
+		@RequestBody final String image
+	) {
+		final Schema.Permission permission = projectService.checkPermissionCanRead(
+			currentUserService.get().getId(),
+			projectId
+		);
+
+		// validate the image
+		if (image == null || image.isEmpty()) {
```

Comment:
I think that this can be done in the parameters area with a "required" flag?

Reply:
I do not think you can make a body required?

Reply:
fixed

Reply:
The body is required by default. It won't let you have an empty body in the post call. I removed this check